app/models/bookings/placement_date.rb

```diff
@@ -39,6 +39,13 @@ class PlacementDate < ApplicationRecord
       presence: true,
       on: :create
 
+    # users manually selecting this only happens when schools are both primary
+    # and secondary, otherwise it's automatically set in the controller
+    validates :supports_subjects,
+      inclusion: [true, false],
+      on: :create,
+      if: -> { bookings_school&.primary_and_secondary? }
```

Comment:
do we need the `on: :create` + `if: -> { bookings_school&.primary_and_secondary? }` conditional here? Doesn't moving setting the `supports_subjects` flag in side the `valid?` conditional in the controller not handle the issue?

Reply:
Also agree, good spot